bc/channel/utils/connectors/bluesky_api/client.py

```diff
@@ -88,6 +88,36 @@ def get_current_time_iso(self) -> str:
         """Get current time in Server Timezone (UTC) and ISO format."""
         return datetime.now(timezone.utc).isoformat()
 
+    def _parse_tags(self, text: str) -> list[RegexMatch]:
+        """
+        Parses hashtags from text.
+
+        This helper function takes a string as input and attempts to extract
+        hashtags from it. If any hashtags are found, they are appended to a
+        list of hashtags. If no hashtags are found, an empty list is returned.
+
+        Args:
+            text (str): The text to parse.
+
+        Returns:
+            list[RegexMatch]: List of matches.
+        """
+        spans = []
+        # reference: https://github.com/bluesky-social/atproto/blob/fbc7e75c402e0c268e7e411353968985eeb4bb06/packages/api/src/rich-text/util.ts#L10
+        # given that our needs of a hashtag is very simple, we can do away with
+        # only parsing alphanumeric characters
+        tag_regex = rb"(?:^|\s)#(?P<tag>[0-9]*[a-zA-Z][a-zA-Z0-9]*)"
+        text_bytes = text.encode("UTF-8")
```

Comment:
If the text is a string, which it appears to be, then there's no need to cast everything to `bytes` objects.
The regex doesn't need the `b` prefix.

Reply:
@johnludwigm Thanks for taking the time to review the PR! I apologize for the late reply.

I'd like to explain the reason behind casting everything to `bytes`: Bluesky strings are internally UTF-8 encoded, and the documentation suggests handling strings in UTF-8 to compute "valid indexes" for elements like Links, Mentions, and Hashtags (you can learn more about it in this [link](https://docs.bsky.app/docs/advanced-guides/post-richtext)). While UTF-8 encoding wouldn't be a problem for plain text posts, we often use emojis in our templates. Emojis can be encoded as multiple characters, leading to incorrect indexes if not handled properly.

I'll create a PR to tweak this helper.

Reply:
oooh right yeah I forgot about it needing to be in UTF-8 specifically